### PR TITLE
Initial dump of filesystem

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,5 +12,4 @@ build/**
 cmake-build-debug/
 
 # ROM
-*.nds
```
Comment (martmists) [inline .gitignore:15]: Why was this removed? If we're switching to naming the file .srl, this should be reflected in all other files that reference it.

File: tools/jsonproc/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+jsonproc
```
Comment (martmists) [inline tools/jsonproc/.gitignore:1]: jsonproc should be another repository and should be included as a submodule instead. Alternatively, add it as required dependency since it seems to have public binary builds that can easily be installed, in which case this should be documented in INSTALL.md.

File: .gitignore (hunk 1 above)
Comment (PikalaxALT) [reply .gitignore:15]: Duplicate entry

File: tools/jsonproc/.gitignore (hunk 2 above)
Comment (PikalaxALT) [reply tools/jsonproc/.gitignore:1]: This requested change is out of scope for this PR.
